src/api/thread-pool.rkt

```diff
@@ -17,6 +18,7 @@
   (format "~a-~a" index (substring replaced 0 (min (string-length replaced) 50))))
 
 (define (run-test index test #:seed seed #:profile profile? #:dir dir)
+  (define bad-id -1) ; TODO move this code to using server.rkt
```

Comment:
Ya, this is one of the first things I would like to do after this PR is see if I can move the nightly `thread-pook.rkt` and `ci.rkt` to using this new `server.rkt`. Moving more data through server code paths in the hope I can fix any bugs it might have sooner than finding them in Odyessy.

The `print-test-result` function expects a table row so I reused `make-improve-result` which takes an `id` and `get-table-data-from-hash` which doesn't use the `id`. 

Lots of potential code to delete after this PR IMO.